### PR TITLE
Chakra UI setup

### LINKED ISSUE #7196: Install Chakra UI
Dependency: `emotion` #7054

### PR DESCRIPTION
**Pull request by pettinarip**
**Note:** branched off from #7179 (depends on `emotion`)

## Description

Initial Chakra UI setup.

Things addressed in this PR:
- Installed Chakra (with Gatsby plugins)
- Overridden default Chakra theme with an initial custom theme
- Reorganized legacy global styles in a new `GlobalStyle` component
- Synced old dark mode state with Chakra color mode state

## Related Issue

Epic #6374 
Fixes #7196

### CODE REVIEW

File: src/@chakra-ui/gatsby-plugin/styles.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const styles = {
+  global: (_props) => ({
+    // TODO: remove these overrides as we adopt the new Design System
```
Comment (samajammin) [inline src/@chakra-ui/gatsby-plugin/styles.ts:3]: A bit more documentation/context on why this is needed to begin with would be helpful.

Comment (samajammin) [reply src/@chakra-ui/gatsby-plugin/styles.ts:3]: e.g. why is both this file & GlobalStyle.tsx needed?

File: src/@chakra-ui/gatsby-plugin/theme.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+// Global style overrides
+import styles from "./styles"
+
+// Foundational style overrides
+import foundations from "./foundations"
+
+const config: ThemeConfig = {
+  cssVarPrefix: "eth",
+  initialColorMode: "light",
+  useSystemColorMode: false,
```
Comment (corwintines) [inline src/@chakra-ui/gatsby-plugin/theme.ts:16]: I think we should use the system color mode personally, I enjoy this feature when I use websites.

Comment (pettinarip) [reply src/@chakra-ui/gatsby-plugin/theme.ts]: Agreed. I've changed it 👍🏼

**Review by corwintines — APPROVED** [review]
LGTM. Tried to break a bunch of pages with no success yet.

**Pull request merged**